components/board/genericlinux/data.go

```diff
@@ -96,16 +98,14 @@ func GetGPIOBoardMappings(modelName string, boardInfoMappings map[string]BoardIn
 // getCompatiblePinDefs returns a list of pin definitions, from the first BoardInformation struct
 // that appears compatible with the machine we're running on.
 func getCompatiblePinDefs(modelName string, boardInfoMappings map[string]BoardInformation) ([]PinDefinition, error) {
-	const compatiblePath = "/proc/device-tree/compatible"
+	var compatibles utils.StringSet
+	var err error
+
+	compatibles, err = rdkutils.GetArchitectureInfo(modelName)
```

Comment:
I'd merge these 4 lines together again. Why declare `compatibles` and `err` separately if you're about to initialize them anyway?